### PR TITLE
Fix Schemas, adds some tests

### PR DESCRIPTION
**Pull request by m4v15**
Addresses #70 #87 #84, relates #86 

- Makes categories and other enum fields in our schemas behave as we want them to #84 
- Add multiple tests to check this, in the event.test.js #70 
- rename category to categories #87 
- loose the id's in the sub docs #84 
- consistently name the foreign id's ```ownerId``` and ```placeId``` #84

### CODE REVIEW

File: src/middleware/mongooseErrorHandler.js [hunk 1]
```diff
@@ -9,14 +9,23 @@ module.exports = (err, req, res, next) => {
...
+      res.boom.badRequest(errMessages.INVALID_ID)
       break
 
     // unhandled mongoose error
     default:
       res.boom.badImplementation(errMessages.UNHANDLED_MONGOOSE)
   }
 }
+
+const extractMongooseMessages = (mongooseErr) => {
```
Comment (m4v15) [inline src/middleware/mongooseErrorHandler.js]: Shall we put this in a helper? I thought maybe not as it's specifically for the mongoose error handler

Comment (mattlub) [reply src/middleware/mongooseErrorHandler.js]: change the comments in this section about the structure of the error?

File: src/middleware/mongooseErrorHandler.js [hunk 2]
```diff
@@ -9,14 +9,23 @@ module.exports = (err, req, res, next) => {
   // }
   switch (err.name) {
     case errNames.MONGOOSE_VALIDATION:
-      res.boom.badRequest(errMessages.VALIDATION_FAILED, err[err.name]) // trying to add more info as 'data' (See boom docs)
+      const reasons = extractMongooseMessages(err)
+      res.boom.badRequest(errMessages.VALIDATION_FAILED, { reasons })// add more info as 'data' (See boom docs)
```
Comment (mattlub) [inline src/middleware/mongooseErrorHandler.js]: do we need the comment now?

File: src/middleware/mongooseErrorHandler.js (hunk 1 above)
Comment (mattlub) [inline src/middleware/mongooseErrorHandler.js]: don't need the brackets here around mongooseErr (line 25)

File: src/middleware/mongooseErrorHandler.js [hunk 3]
```diff
@@ -9,14 +9,23 @@ module.exports = (err, req, res, next) => {
...
 
     // unhandled mongoose error
     default:
       res.boom.badImplementation(errMessages.UNHANDLED_MONGOOSE)
   }
 }
+
+const extractMongooseMessages = (mongooseErr) => {
+  const errorMessages = []
+  Object.keys(mongooseErr.errors).forEach((singleError) => {
```
Comment (mattlub) [inline src/middleware/mongooseErrorHandler.js]: map instead?

**Review by mattlub — CHANGES_REQUESTED** [review]

File: src/middleware/mongooseErrorHandler.js (hunk 2 above)
Comment (m4v15) [reply src/middleware/mongooseErrorHandler.js]: No, got rid

File: src/middleware/mongooseErrorHandler.js (hunk 1 above)
Comment (m4v15) [reply src/middleware/mongooseErrorHandler.js]: done

Comment (m4v15) [reply src/middleware/mongooseErrorHandler.js]: soz

File: src/middleware/mongooseErrorHandler.js (hunk 3 above)
Comment (m4v15) [reply src/middleware/mongooseErrorHandler.js]: 👍

**Comment by m4v15** [comment]
@mattlub Addressed your issues

**Review by mattlub — APPROVED** [review]

**Review by des-des — APPROVED** [review]

**Pull request merged**